lib/kernel/src/group.erl

```diff
@@ -886,6 +894,41 @@ get_chars_echo_off1(Drv, Shell) ->
             exit(R)
     end.
 
+format_expression(Cont) ->
+    FormatingCommand = application:get_env(stdlib, format_shell_func),
+    Buffer = edlin:current_line(Cont),
+    try
+        case FormatingCommand of
+            {ok, {M,F}} when is_atom(M), is_atom(F) ->
+                    M:F(Buffer);
+            {ok, FormatingCommand1} when is_list(FormatingCommand1) ->
+                format_expression1(Buffer, FormatingCommand1)
+        end
+    catch
```

Comment:
If the formatting function crashes, we should log and possibly remove the current formatter?